sysdig/resource_sysdig_secure_organization.go

```diff
@@ -0,0 +1,154 @@
+package sysdig
+
+import (
+	"context"
+	proto "github.com/draios/protorepo/cloudauth/go"
+	v2 "github.com/draios/terraform-provider-sysdig/sysdig/internal/client/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"time"
+)
+
+func resourceSysdigSecureOrganization() *schema.Resource {
+	timeout := 5 * time.Minute
+
+	return &schema.Resource{
+		CreateContext: resourceSysdigSecureOrganizationCreate,
+		DeleteContext: resourceSysdigSecureOrganizationDelete,
+		ReadContext:   resourceSysdigSecureOrganizationRead,
+		UpdateContext: resourceSysdigSecureOrganizationUpdate,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(timeout),
+			Update: schema.DefaultTimeout(timeout),
+			Read:   schema.DefaultTimeout(timeout),
+			Delete: schema.DefaultTimeout(timeout),
+		},
+		Schema: map[string]*schema.Schema{
+			"cloud_provider_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"cloud_provider_type": {
+				Type:         schema.TypeInt,
```

Comment:
Even though the provider type is fetched from an enum, the schema expects a string type. Even the single account API's schema expects string type.
https://github.com/draios/secure-backend/blob/main/docs/api/cloudauth/v1/swagger.yaml#L377

Reply:
we can do a .String() on the enum as below